### PR TITLE
refactor(app-shell): render theme based on feature flag

### PR DESCRIPTION
**Pull request by emmenko**
Prepare the theme to be switched based on a feature flag. This will be used initially for testing purposes only.

### CODE REVIEW

File: packages/application-shell/src/components/application-shell/application-shell.tsx [hunk 1]
```diff
@@ -243,6 +243,7 @@ export const RestrictedApplication = <
                     defaultFlags={props.defaultFeatureFlags}
                   >
                     <>
+                      <ThemeSwitcher />
```
Comment (emmenko) [inline packages/application-shell/src/components/application-shell/application-shell.tsx:246]: Moved here to be "within" the feature flags provider.

**Comment by chloe0592** [comment]
Thank you for your work! I was wondering how we do feature flags. :)

**Review by chloe0592 — APPROVED** [review]

File: packages/application-shell/src/components/theme-switcher/theme-switcher.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { ThemeProvider } from '@commercetools-uikit/design-system';
+import { useFeatureToggle } from '@flopflip/react-broadcast';
+import { UI_REDESIGN } from '../../feature-toggles';
+
+const ThemeSwitcher = () => {
```
Comment (CarlosCortizasCT) [inline packages/application-shell/src/components/theme-switcher/theme-switcher.tsx:5]: **nit**: I would expect a component with this name to actually implement a theme selector. Perhaps we will have it in the future when we implement dark theme.

Comment (emmenko) [reply packages/application-shell/src/components/theme-switcher/theme-switcher.tsx:5]: Yeah we will need to refactor things again once we get there.

**Review by Rhotimee — APPROVED** [review]
🙌🏽

**Pull request merged**